grails-app/services/org/pih/warehouse/order/PurchaseOrderIdentifierService.groovy

```diff
@@ -0,0 +1,70 @@
+package org.pih.warehouse.order
+
+import org.apache.commons.lang.StringUtils
+
+import org.pih.warehouse.core.IdentifierGeneratorTypeCode
+import org.pih.warehouse.core.IdentifierService
+import org.pih.warehouse.core.IdentifierTypeCode
+import org.pih.warehouse.core.Organization
+import org.pih.warehouse.core.identification.IdentifierGeneratorParams
+import org.pih.warehouse.shipping.Shipment
+
+// TODO: We should be able to move this sequential logic into the identifier service. That way we can support sequential
+//       id generation more generally. We'd need an abstract method there for fetching the next sequence number to use,
+//       and a new ${sequence} config option to know where to put the sequence.
+class PurchaseOrderIdentifierService extends IdentifierService {
```

Comment:
A majority of this code is just copied out from IdentifierService. I couldn't get around to refactoring this as a part of this change.

See the TODO here for how we could refactor in the future.